src/OmniSharp.Host/Program.cs

```diff
@@ -61,7 +61,7 @@ private static int Run(string[] args)
             var omnisharpApp = new CommandLineApplication(throwOnUnexpectedArg: false);
             omnisharpApp.HelpOption("-? | -h | --help");
 
-            var applicationRootOption = omnisharpApp.Option("-s | --source", "Solution, project file or directory for OmniSharp to point at (defaults to current directory).", CommandOptionType.SingleValue);
```

Comment:
From what I could tell, all of our project systems assume that `OmniSharpEnvironment.Path` is a directory. If OmniSharp had been launched with anything other than a path to a directory or a solution file, it wouldn't have worked. Let me know if I've read this incorrectly.